src/dfx/src/commands/extension/list.rs

```diff
@@ -1,22 +1,51 @@
 use crate::lib::environment::Environment;
 use crate::lib::error::DfxResult;
+use clap::Parser;
 use std::io::Write;
+use tokio::runtime::Runtime;
+use url::Url;
 
-pub fn exec(env: &dyn Environment) -> DfxResult<()> {
+#[derive(Parser)]
+pub struct ListOpts {
+    /// Specifies to list the installed extensions.
+    #[arg(long, conflicts_with("catalog_url"))]
+    installed: bool,
```

Comment:
This would be a breaking change, but also inconsistent with other tools.

First, it would be a breaking change: `dfx extension list` today lists the installed extensions.  This would make it instead list the available extensions.

Second, it's inconsistent with `dfxvm` as well as with the tools from which we took inspiration:
- `dfxvm list`: `List installed versions of dfx`
- `rustup toolchain list`: `List installed toolchains`
- `brew list`: `List all installed formulae and casks`

Suggestion: change to `dfx extension list --available`

Reply:
Thanks, that makes great sense. Fixed at https://github.com/dfinity/sdk/pull/3896/commits/3791d56c3329f2fb92e9bb701cf296d0ac2cc9bb.